File: Reflection-Beauty-Salon-Backend-main/src/main/java/com/symbiosis/reflection/service/UpdateServiceService.java
package com.symbiosis.reflection.service;

import java.util.List;
import com.symbiosis.reflection.model.UpdateService;

public interface UpdateServiceService {
    
    UpdateService saveService(UpdateService service);
    
    List<UpdateService> getAllServices();
    
    UpdateService getServiceById(int id);
    
    void deleteService(int id);
    
    UpdateService updateService(int id, UpdateService service);
    
    List<UpdateService> getServicesByCategory(String category);
    
    UpdateService getServiceByName(String serviceName);

	long getCount();
}

File: Reflection-Beauty-Salon-Frontend-main/src/app/service/updateservice.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Updateservice } from '../models/updateservice';

@Injectable({
  providedIn: 'root'
})
export class UpdateserviceService {
serverurl="http://localhost:9000/updateservices"
  constructor(private http:HttpClient) { }

  getAllServices():Observable<Updateservice[]>
  {
    return this.http.get<Updateservice[]>(this.serverurl)
  }

  addService(service:Updateservice):Observable<Updateservice>
  {
    return this.http.post<Updateservice>(this.serverurl,service)
  }

  getServiceById(i:any):Observable<Updateservice>
  {
    return this.http.get<Updateservice>(this.serverurl+"/"+i)
  }

  deleteService(i:any):Observable<Updateservice>
  {
    return this.http.delete<Updateservice>(this.serverurl+"/"+i)
  }

  updateService(i:any,ups:Updateservice):Observable<Updateservice>
  {
    return this.http.put<Updateservice>(this.serverurl+"/"+i,ups)
  }

  getServicesByCategory():Observable<Updateservice[]>
  {
    return this.http.get<Updateservice[]>(this.serverurl)
  }

  getServiceByName():Observable<Updateservice[]>
  {
    return this.http.get<Updateservice[]>(this.serverurl)
  }
}
